seacatauth/authn/webauthn/service.py

```diff
@@ -0,0 +1,422 @@
+		upsertor.set("ct", verified_registration.credential_type.value)
+		upsertor.set("ao", verified_registration.attestation_object)
+		upsertor.set("name", name)
+
+		wacid = await upsertor.execute()
+		L.log(asab.LOG_NOTICE, "WebAuthn credential created", struct_data={"wacid": wacid})
+
+
+	async def get_webauthn_credential(self, webauthn_credential_id):
+		"""
+		Get WebAuthn credential detail by its ID
+		"""
+		return await self.StorageService.get(self.WebAuthnCredentialCollection, webauthn_credential_id)
+
+
+	async def get_webauthn_credentials_by_user(self, credentials_id: str):
+		"""
+		Get all WebAuthn credentials of a specified user
+		"""
+		collection = self.StorageService.Database[self.WebAuthnCredentialCollection]
+
+		query_filter = {"cid": credentials_id}
+		cursor = collection.find(query_filter)
+
+		cursor.sort("_c", -1)
+
+		wa_credentials = []
+		async for resource_dict in cursor:
+			wa_credentials.append(resource_dict)
+
+		return wa_credentials
+
+
+	async def update_webauthn_credential(
+		self, webauthn_credential_id: bytes, *,
+		credentials_id: str = None,
+		sign_count: int = None,
+		name: str = None,
+		last_login: datetime.datetime = None,
+	):
+		"""
+		Update WebAuthn credential
+
+		Only allows updating key name, last login time and sign count.
+
+		If credentials_id is specified, ensure that it matches the database entry.
+		"""
+		wa_credential = await self.get_webauthn_credential(webauthn_credential_id)
+
+		if credentials_id is not None:
+			if credentials_id != wa_credential["cid"]:
+				raise KeyError("WebAuthn credential not found", {
+					"wacid": webauthn_credential_id,
+					"cid": credentials_id
+				})
+
+		upsertor = self.StorageService.upsertor(
+			self.WebAuthnCredentialCollection,
+			obj_id=webauthn_credential_id,
+			version=wa_credential["_v"]
+		)
+
+		if sign_count is not None:
+			upsertor.set("sc", sign_count)
+
+		if name is not None:
+			upsertor.set("name", name)
+
+		if last_login is not None:
+			upsertor.set("ll", last_login)
+
+		await upsertor.execute()
+		L.log(asab.LOG_NOTICE, "WebAuthn credential updated", struct_data={
+			"wacid": webauthn_credential_id,
+		})
+
+
+	async def delete_webauthn_credential(self, webauthn_credential_id: bytes, credentials_id: str = None):
+		"""
+		Delete WebAuthn credential by its ID.
+
+		If credentials_id is specified, ensure that it matches the database entry.
+		"""
+
+		if credentials_id is not None:
+			wa_credential = await self.get_webauthn_credential(webauthn_credential_id)
+			if credentials_id != wa_credential["cid"]:
+				raise KeyError("WebAuthn credential not found", {
+					"wacid": webauthn_credential_id,
+					"cid": credentials_id
+				})
+
+		await self.StorageService.delete(self.WebAuthnCredentialCollection, webauthn_credential_id)
+		L.log(asab.LOG_NOTICE, "WebAuthn credential deleted", struct_data={"wacid": webauthn_credential_id})
+
+
+	async def delete_webauthn_credentials_by_user(self, credentials_id: str):
```

Comment:
`delete_webauthn_all_credentials` ... we don't use "user" name (and I understand the overlap with FIDO2 credential)